src/components/List/Context.ts

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext } from 'react';
+
+export enum BorderStyle {
+  Solid = 'solid',
+  Dashed = 'dashed',
+}
+
+export const ListContext = createContext<
+  | {
+      borderStyle: BorderStyle;
+    }
+  | undefined
+>(undefined);
```

Comment:
Since we want to use `BorderStyle.Solid` as a default when no style is specified, we can simplify this to just

```suggestion
export const ListContext = createContext({
  borderStyle: BorderStyle;
});
```

It doesnt really make much differece here, since this will only kick in if we dont have a context provider higher up in the tree. Because of how we use this context, that will never be the case. But regardless, I think its good to set sane defaults where it makes sense. That also helps improve later code, since code is often copied from earlier implementations.